client/state/selectors/is-site-checklist-complete.js

```diff
@@ -50,5 +52,14 @@ export default function isSiteChecklistComplete( state, siteId ) {
 		return false;
 	};
 
-	return siteChecklist.tasks.every( isTaskComplete );
+	const taskList = getTaskList( {
+		taskStatuses: siteChecklist.tasks,
+		phase2: siteChecklist.phase2,
+		siteVerticals: siteChecklist.verticals,
+		siteSegment: siteChecklist.siteSegment,
+	} ).getAll();
+	const taskUrls = getTaskUrls( state, siteId );
+	const hasUrl = task => ! ( taskUrls.hasOwnProperty( task.id ) && ! taskUrls[ task.id ] );
```

Comment:
Nitpick: I may be missing a particular detail here, but the following looks like a much simpler alternative:

```js
const hasUrl = task => task.id in taskUrls && taskUrls[ task.id ];
```

Reply:
Ah, using `in` is probably clearer you're right! The reason for the first check is that I'm trying to make sure that the object has a property and then check if its value is `null` or not. It's also badly named as I want the function to return `true` when the property's value is `null` and not when it is a URL string. I've made some tweaks. The logic could be changed to something like:

```
const hasNotUrl = task => ! task.id in taskUrls || taskUrls[ task.id ];
```

If we thought that was clearer still?

Reply:
Ah, thank you for clarifying! The desired logic is a bit confusing, though. Your proposed alternative would:

- Return `true` when there is no entry in `taskUrls`
- Return `true` when there is an entry in `taskUrls` and that entry has a truthy value

It would return `false` when there is an entry in `taskUrls` and that entry has a falsey value. Falsey values in JS include `false`, `null`, and `undefined`, but they also include `0`, `''` and `NaN`.

Is that the desired behaviour?

Reply:
Yeah that's the desired behaviour. The logic is a bit odd, but certain tasks need a URL in order to be completed, and there are other tasks in the list where a URL is not needed. 

If a task would normally need a URL then it would it be a property in the `taskURLs` object, so then we want it filtered from the task list if the value is in `taskURLs` is falsey. 

Otherwise, if it's not a property of the `taskURLs` object then it doesn't need a URL and shouldn't be filtered from the task list.

The way the `taskURLs` object is created, the value should either be a non-empty string or `null`, but if some other falsey value gets in there too, then that would be fine as it's not a valid URL, and we would want to filter that to prevent an error.

Reply:
Sounds good to me, then, thank you again for the explanation! 👍

Reply:
> It's also badly named as I want the function to return true when the property's value is null and not when it is a URL string.

I've just noticed that this is not true! It's the other way around, and so I think I was right with naming it `hasUrl` in the first place. That's what happens when you step away from some complex logic for a while!